app/src/main/java/pl/com/salwa/unohouse/unohouseandroid/activities/LoginActivity.java

```diff
@@ -0,0 +1,210 @@
+
+import pl.com.salwa.unohouse.unohouseandroid.models.AuthenticationResponse;
+import pl.com.salwa.unohouse.unohouseandroid.models.Credentials;
+import pl.com.salwa.unohouse.unohouseandroid.networking.UnoHouseAPI;
+import pl.com.salwa.unohouse.unohouseandroid.networking.UnoHouseAPIClient;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * A login screen that offers login via login/password.
+ */
+public class LoginActivity extends AppCompatActivity {
+
+    public static final String APP_TAG = "Uno";
+
+    // UI references.
+    private EditText mLoginView;
+    private EditText mPasswordView;
+    private View mProgressView;
+    private View mLoginFormView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        // Set up the login form.
+        mLoginView = (EditText) findViewById(R.id.login);
+        mPasswordView = (EditText) findViewById(R.id.password);
+
+        Button signInButton = (Button) findViewById(R.id.login_button);
+        signInButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                attemptLogin();
+            }
+        });
+
+        mLoginFormView = findViewById(R.id.login_form);
+        mProgressView = findViewById(R.id.login_progress);
+
+        checkAppVersion();
+    }
+
+    private void checkAppVersion() {
+        showProgress(true);
+
+        UnoHouseAPI apiService = UnoHouseAPIClient.getClient().create(UnoHouseAPI.class);
+
+        Call<String> call = apiService.latestVersion();
+        call.enqueue(new Callback<String>() {
+
+            @Override
+            public void onResponse(Call<String> call, Response<String> response) {
+                int statusCode = response.code();
+                Log.d(APP_TAG, statusCode + " status code");
+                String s = response.body();
+
+                    Log.d(APP_TAG, s + " result");
+                showProgress(false);
+            }
+
+            @Override
+            public void onFailure(Call<String> call, Throwable t) {
+                Log.d(APP_TAG, t.toString());
+                showProgress(false);
+            }
+        });
+    }
+
+    /**
+     * Attempts to sign in.
+     * If there are form errors (invalid login, missing fields, etc.), the
+     * errors are presented and no actual login attempt is made.
+     */
+    private void attemptLogin() {
+//        if (mAuthTask != null) {
+//            return;
+//        }
+
+        // Reset errors.
+        mLoginView.setError(null);
+        mPasswordView.setError(null);
+
+        // Store values at the time of the login attempt.
+        String login = mLoginView.getText().toString();
+        String password = mPasswordView.getText().toString();
+
+        boolean cancel = false;
+        View focusView = null;
+
```

Comment:
jeżeli nei ma takiej potrzeby, to zmienne inicjujemy na początku funkcji/metody. Wyjątkiem są local variables dla pętli, warunków itp.